### ISSUE TITLE
SERVER AND USER ISSUE: Client blocked on "Entering World..."

### ISSUE THREAD

**Issue by Redblueflame** [StrangeLoopGames/EcoIssues#6882]
Eco version: 7.3.1 (Latest)
Issue description:
     Sometimes (every day), all the users are stuck on "Entering World...", but only on this server, the users tried to connect to another server and works. The only way to resolve it is to restart the server.
Steps to reproduce:

- Start the server
- Wait...
- The user will get stuck on "Entering World..."

Normal way:
- Start the server
- Access normally to the server.

**Comment by THEGOBBLER** [comment]
Also getting this. Server has been up for 3 days, it started on day #2 and has happened two times in 24 hours so far. 

Prior to it happening people started experiencing high pings 3-600ms.

**Comment by MarjieAVolk** [comment]
Related to https://github.com/StrangeLoopGames/EcoIssues/issues/4669 if that's the case

**Comment by geeknessfr** [comment]
Hi,
We get the same issue on our dedicated server (15 - 20 players per day), we had to restart server at least 1 to 4 times per day .... We first believed that was caused by a Mod, but same issue happened without any mod.

This happen when voting for a law, or just building, digging.... etc... I can't say exactly what is causing that, looks like a thread is crashing but without any error message. Symptoms in game are variable, first, craft tables doesn't craft anymore, the time showed is frozen and they don't take any materials in stockpiles or chest. Digging, Building, etc... not responding as fast as before... and in the mean time other players can't join the server and are in an infinite loop of "Entering World..."

**Comment by craig-jong** [comment]
should be fixed in 7.3.2 - please reopen it if it continues.

**Comment by gabeux** [comment]
Thank you Marjie for reopening and looking into this!
Here's a server crashdump related to the laws issue. Our admins cleaned the laws, so basically only a pollution law was in effect, and a district law was proposed. We highly suspect it's the district laws.

[ps: renamed file extension from .ecodmp to .txt so github accepts it]

[Crash_AggregateException_04095623.txt](https://github.com/StrangeLoopGames/EcoIssues/files/1891893/Crash_AggregateException_04095623.txt)

**Comment by Rev8** [comment]
Hello Marjie, here's the file you requested from Gabeux. Hope it helps finding a solution!

The file is too big to upload directly to here (10MB limit). Here's a direct link.
https://drive.google.com/open?id=1pTCbY0ve7P77stnrAa3hesKAYPMIxk_2

**Comment by MarjieAVolk** [comment]
Thank you!

**Comment by MarjieAVolk** [comment]
Okay guys, I have found a fix for the dump posted by @gabeux.  It's unclear whether this is related to the original 'stuck on entering world' issue yet, as I have not repro-ed that.

**Comment by MarjieAVolk** [comment]
How is the stuck on entering world problem looking in 7.3.3?  Judging by the hearts, sounds like it might be better now?  :P

**Comment by Sekor2600** [comment]
I did have the Problem again today

**Comment by geeknessfr** [comment]
For our server, it looks better, but now this issue is caused by a Stat issue (Sorry i didn't open a new issue for that.) ...

![image](https://user-images.githubusercontent.com/28137335/38744079-1f8ce676-3f41-11e8-8345-7e0ba25ab51d.png)
![image](https://user-images.githubusercontent.com/28137335/38744095-2b3b6376-3f41-11e8-9daa-6cfa8558550f.png)
![image](https://user-images.githubusercontent.com/28137335/38744115-3a0bd8c2-3f41-11e8-8267-46fff2d8ac85.png)

We had to disables stats on our server to resolve this issue (in 7.3.2 and not tested yet on 7.3.3)

Yesterday one of our user had also the issue mentionned here : #6970

**Comment by MarjieAVolk** [comment]
The ThreadAbortException problem is tracked in https://github.com/StrangeLoopGames/EcoIssues/issues/6996

**Comment by MarjieAVolk** [comment]
So you're saying the thread abort thing is related to people being stuck on entering world too?

**Comment by Sekor2600** [comment]
The only Thing that i can do ist wait till entering World ist finish. But then the only Thing that i can do ist place blocks. All other kicks with inactiviy. But no Lags. i can drive Execuvator but no usw with excavator shovel. It's like the world is frozen and you can only use 2 things. All other Player joinen/leaven

**Comment by Sekor2600** [comment]
Ok did ask Serveradmin. He did have the same ThreadAbort errors. But dont know that it is related to Stuck in Entering

**Comment by MarjieAVolk** [comment]
Okay thank you

**Comment by geeknessfr** [comment]
@MarjieAVolk When the thread abort thing happened, all the players couldn't join the server while we didn't restart it, yes... But before, in 7.3.1, the Entering World thing happened without any error on the server side.

**Comment by Pretmaker** [comment]
Noticing the same issue. Once you get kicked for inactivity it seems like the server kills itself.

Server dumps also show thread aborted.

**Comment by MarjieAVolk** [comment]
Well guys, since we've reworked server connection in 7.4, I guess we're going to close this one as obsolete now.

Had this gotten better during the last few weeks?  I didn't see many recent complaints about it.
